### PR TITLE
Add production ready and GA sections to roadmap

### PR DESCRIPTION
**Pull request by stefanprodan**
Changes:
- Add draft GA roadmap
- Add production ready section to clarify which APIs and controllers are considered production ready
- Clarify that image update automation based on timestamps found inside image layers has been deprecated

Preview here: https://toolkit.fluxcd.io/roadmap/

### CODE REVIEW

File: docs/roadmap/index.md [hunk 1]
```diff
@@ -1,15 +1,47 @@
 # Roadmap
 
-In our planning discussions we identified broad three areas of work:
+!!!hint "Production Ready"
+    The Flux custom resource definitions which are at `v1beta1` and `v2beta1`
+    and their controllers are considered stable and production ready.
+    Going forward, changes to the stable CRDs will be accompanied by a conversion mechanism.
```
Comment (hiddeco) [inline docs/roadmap/index.md:6]:
```suggestion
    Going forward, breaking changes to the stable CRDs will be accompanied by a conversion mechanism.
```

maybe? As we make changes that are non-breaking without offering a conversion mechanism.

File: docs/roadmap/index.md [hunk 2]
```diff
@@ -1,15 +1,47 @@
 # Roadmap
 
-In our planning discussions we identified broad three areas of work:
+!!!hint "Production Ready"
```
Comment (hiddeco) [inline docs/roadmap/index.md:3]:
```suggestion
!!!hint "Production readiness"
```

File: docs/roadmap/index.md (hunk 1 above)
Comment (stefanprodan) [reply docs/roadmap/index.md]: good point! thanks

**Review by hiddeco — APPROVED** [review]
LGTM, thank you @stefanprodan :100:

File: docs/roadmap/index.md [hunk 3]
```diff
@@ -1,15 +1,48 @@
...
+- `kustomize.toolkit.fluxcd.io/v1beta1`
+- `notification.toolkit.fluxcd.io/v1beta1`
+- `helm.toolkit.fluxcd.io/v2beta1`
+
+## The road to Flux v2 GA
+
+In our planning discussions we identified these areas of work:
+
+- Stabilize the image automation APIs
+    * Review the spec of `ImageRegistry`, `ImagePolicy` and `ImageUpdateAutomation`
```
Comment (squaremo) [inline docs/roadmap/index.md:28]:
```suggestion
    * Review the spec of `ImageRepository`, `ImagePolicy` and `ImageUpdateAutomation`
```

**Review by squaremo — COMMENTED** [review]
I think it's also worth saying that planning is ongoing -- this certainly happened with the v1 parity milestone, we changed our idea of what was in scope and discovered problems we didn't know at the start.

**Pull request merged**